features/cellular/framework/AT/AT_CellularStack.cpp

```diff
@@ -307,7 +307,12 @@ nsapi_size_or_error_t AT_CellularStack::socket_recvfrom(nsapi_socket_t handle, S
 
         _at.unlock();
         if (ret_val != NSAPI_ERROR_OK) {
-            tr_error("Socket %d create %s error %d", find_socket_index(socket), addr->get_ip_address(), ret_val);
+            if(addr){
+                tr_error("Socket %d create %s error %d", find_socket_index(socket), addr->get_ip_address(), ret_val);
```

Comment:
This is not right.
The `recvfrom()` is not yet called, so the address cannot be correct anyway.

I would remove the whole `if (addr)` and use the second `tr_error()` format.

Reply:
I assume it is an input&output parameter for UDP, so that I kept original prints if it is available, could you help to check?

Reply:
Got it, thanks for the comments, will update accordingly.